synapse/handlers/stats.py

```diff
@@ -52,4 +59,324 @@ def __init__(self, hs):
+                    room_stats_delta["left_members"] = (
+                        room_stats_delta.get("left_members", 0) - 1
+                    )
+                elif prev_membership == Membership.BAN:
+                    room_stats_delta["banned_members"] = (
+                        room_stats_delta.get("banned_members", 0) - 1
+                    )
+                else:
+                    err = "%s is not a valid prev_membership" % (repr(prev_membership),)
+                    logger.error(err)
+                    raise ValueError(err)
+
+                if membership == Membership.JOIN:
+                    room_stats_delta["joined_members"] = (
+                        room_stats_delta.get("joined_members", 0) + 1
+                    )
+                elif membership == Membership.INVITE:
+                    room_stats_delta["invited_members"] = (
+                        room_stats_delta.get("invited_members", 0) + 1
+                    )
+                elif membership == Membership.LEAVE:
+                    room_stats_delta["left_members"] = (
+                        room_stats_delta.get("left_members", 0) + 1
+                    )
+                elif membership == Membership.BAN:
+                    room_stats_delta["banned_members"] = (
+                        room_stats_delta.get("banned_members", 0) + 1
+                    )
+                else:
+                    err = "%s is not a valid membership" % (repr(membership),)
+                    logger.error(err)
+                    raise ValueError(err)
+
+                user_id = state_key
+                if self.is_mine_id(user_id) and membership in (
+                    Membership.JOIN,
+                    Membership.LEAVE,
+                ):
+                    # update user_stats as it's one of our users
+                    public = yield self._is_public_room(room_id)
+
+                    field = "public_rooms" if public else "private_rooms"
+                    delta = +1 if membership == Membership.JOIN else -1
+
+                    yield self.store.update_stats_delta(
+                        now, "user", user_id, {field: delta}
+                    )
+
+            elif typ == EventTypes.Create:
+                # Newly created room. Add it with all blank portions.
+                yield self.store.update_room_state(
+                    room_id,
+                    {
+                        "join_rules": None,
+                        "history_visibility": None,
+                        "encryption": None,
+                        "name": None,
+                        "topic": None,
+                        "avatar": None,
+                        "canonical_alias": None,
+                    },
+                )
+
+                room_stats_complete = True
+
+            elif typ == EventTypes.JoinRules:
+                old_room_state = yield self.store.get_room_state(room_id)
```

Comment:
This is probably going to be more up to date room state?

Reply:
As discussed IRL (I think – lmk if mistaken), this should be fine, as the line underneath is responsible for updating the `room_stats_state` table upon which this (commented) line depends.